File: core/test/src/testFixtures/java/ir/composenews/core_test/dispatcher/TestDispatcherProvider.java
package ir.composenews.core_test.dispatcher;

import ir.composenews.base.dispatcher.DispatcherProvider;
import kotlinx.coroutines.CoroutineDispatcher;
import kotlinx.coroutines.test.TestCoroutineDispatchersKt;
import kotlinx.coroutines.test.TestCoroutineScheduler;
import kotlinx.coroutines.test.TestDispatcher;

public final class TestDispatcherProvider implements DispatcherProvider {
    private final TestDispatcher ui;
    private final TestDispatcher io;
    private final TestDispatcher bg;

    public TestDispatcherProvider()  {
       this(new TestCoroutineScheduler());
    }

    public TestDispatcherProvider(TestCoroutineScheduler testScheduler) {
        ui = TestCoroutineDispatchersKt.UnconfinedTestDispatcher(testScheduler, null);
        io = TestCoroutineDispatchersKt.UnconfinedTestDispatcher(testScheduler, null);
        bg = TestCoroutineDispatchersKt.UnconfinedTestDispatcher(testScheduler, null);
    }

    @Override
    public CoroutineDispatcher getUi() {
        return ui;
    }

    @Override
    public CoroutineDispatcher getIo() {
        return io;
    }

    @Override
    public CoroutineDispatcher getBg() {
        return bg;
    }
}

File: core/test/src/testFixtures/java/ir/composenews/core_test/dispatcher/TestDispatcherProvider.kt
package ir.composenews.core_test.dispatcher

import ir.composenews.base.dispatcher.DispatcherProvider
import kotlinx.coroutines.test.TestCoroutineScheduler
import kotlinx.coroutines.test.TestDispatcher
import kotlinx.coroutines.test.UnconfinedTestDispatcher

class TestDispatcherProvider(testScheduler: TestCoroutineScheduler) : DispatcherProvider {
    override val ui: TestDispatcher = UnconfinedTestDispatcher(testScheduler)
    override val io: TestDispatcher = UnconfinedTestDispatcher(testScheduler)
    override val bg: TestDispatcher = UnconfinedTestDispatcher(testScheduler)
}
